src/discord/cmds/slash/link.js

```diff
@@ -1,70 +1,86 @@
-const noMatch = createError('**Discord does not match!**');
-const invalidIGN = createError('**Invalid Username!**');
+module.exports = {
+  name: 'link',
+  desc: 'Link your account',
+  options: [{ type: 'string', name: 'ign', desc: 'Enter your IGN', required: true }],
 
-module.exports =
-{
-    name: 'link',
-    desc: 'Link your account',
-    options: [
-        { type: 'string', name: 'ign', desc: 'Enter your IGN', required: true }
-    ],
-
-    async execute(interaction) {
-        await interaction.deferReply();
-
-        const input = interaction.options.getString('ign');
-        const check = await getEmoji('check');
-        const plus = await getEmoji('plus');
-        const minus = await getEmoji('minus');
-
-        try {
-            const player = await getPlayer(input);
-            const discord = await getDiscord(player.uuid);
-            if (!discord) return interaction.followUp({ embeds: [notLinked] });
-            if (interaction.user.username !== discord.toLowerCase()) return interaction.followUp({ embeds: [noMatch] });
-
-            await Link.create({ uuid: player.uuid, dcid: interaction.user.id })
-                .catch((e) => {
-                    if (e.code === 11000) console.log('playersLinked: Duplicate Key!');
-                });
-
-            await interaction.member.setNickname(player.nickname)
-                .catch(e => {
-                    if (e.message.includes('Missing Permissions')) interaction.followUp({ embeds: [createMsg({ color: 'FFD800', desc: '**I don\'t have permission to change your nickname!**' })] });
-                });
-
-            const { addedRoles, removedRoles } = await updateRoles(interaction.member, player, true);
-
-            let desc;
-            if (addedRoles.length > 0 && removedRoles.length > 0) {
-                desc = `${check} **Account linked!**\n_ _\n`;
-                desc += `${addedRoles.map(roleID => `${plus} <@&${roleID}>`).join('\n')}\n_ _\n`;
-                desc += `${removedRoles.map(roleID => `${minus} <@&${roleID}>`).join('\n')}`;
-            }
-            else if (addedRoles.length > 0) {
-                desc = `${check} **Account linked!**\n_ _\n`;
-                desc += `${addedRoles.map(roleID => `${plus} <@&${roleID}>`).join('\n')}\n_ _`;
-            }
-            else if (removedRoles.length > 0) {
-                desc = `${check} **Account linked!**\n_ _\n`;
-                desc += `${removedRoles.map(roleID => `${minus} <@&${roleID}>`).join('\n')}\n_ _`;
-            }
-            else {
-                desc = `${check} **Account linked!**`;
-            }
-
-            return interaction.followUp({ embeds: [createMsg({ desc })] });
+  async execute(interaction) {
+    await interaction.deferReply();
+    const input = interaction.options.getString('ign');
+    const application = await interaction.client.application.fetch();
+    const emojis = await application.emojis.fetch();
+    const check = emojis.find((emoji) => {
```

Comment:
Before you were using native discord API requests. This is bad because there is no caching other stuff. Using discord.js allows for caching and making this effectivly one api request for all 3 emojis and not a request for each emoji

Reply:
i see. 

what if instead of getting rid of getEmoji() entirely, we do something like:

`async function getEmoji(application, name)
{
  const emojis = await application.emojis.fetch();
  return emojis.find((emoji) => name === emoji.name)
}`
and then
`const application = await interaction.client.application.fetch();
const check = await application.getEmoji('check');`

Reply:
Well that works but it still makes the issue of your defining a function and adding an extra set of code

Reply:
and so you would prefer to add 3 lines every single time an emoji is referenced? ;-;

```
    const minus = emojis.find((emoji) => {
      return 'minus' === emoji.name;
    });
```

Reply:
Well there is a better way to do it but that's not currently possible because the requirements for that are not implemented yet